account_journal_general_sequence/hooks.py

```diff
@@ -0,0 +1,22 @@
+from odoo import SUPERUSER_ID, _, api
+
+
+def pre_init_companies(cr):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    cmps = env["res.company"].search([])
+    for company in cmps:
```

Comment:
This is only going to be executed at module install time. It will not run on update. Imagine this story:

1. Database contains company 1 and company 2.
2. Admin installs this module. It creates sequences for companies 1 and 2.
3. Admin creates company 3.

With this implementation, Company 3 will never have a sequence. That's why the previous implementation was more sound.